core/src/main/scala/io/qbeast/core/model/Block.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Qbeast Analytics, S.L.
+ */
+package io.qbeast.core.model
+
+/**
+ * Block represents the index data stored as a continous media in a index file.
+ * It also provides the necessary metadada to maintain the index structure.
+ *
+ * @param file the index file
+ * @param range the range of the index file that stores the data
+ * @param cubeId the cube identifier
+ * @param state the block state
+ * @param minWeight the minimum weight of the block element
+ * @param maxWeight the maximum weight of the block element
+ */
+final case class Block(
```

Comment:
Just a good habit from the Java world. If you want to expose some classes as public API then it is a good idea to make it final to avoid breaking the client code by adding some members which the client already implemented in their subclass. For now qbeastCore is a separate project with its own distributable, so this is public API. To be honest I am not fond of the current approach with the core library, so in the future we can hide this and similar classes if we merge qbeastCore with qbeastSpark, but it is out of the scope now.